pkg/middleware/csrf.go

```diff
@@ -27,12 +27,17 @@ func CSRF(loginCookieName string) func(http.Handler) http.Handler {
 				}
 			}
 			// Otherwise - verify that Origin matches the server origin
-			host := strings.Split(r.Host, ":")[0]
+			host, _, err := net.SplitHostPort(r.Host)
```

Comment:
Actually this requires a port or will error, so I probably do need the "withdefaults" I guess, and need the cfg to know the port I imagine...